### ISSUE TITLE
Add status_on_error option to RateLimit filter

### ISSUE THREAD

**Issue by AleksanderBrzozowski** [envoyproxy/envoy#27423]
*Title*: *Add status_on_error option to RateLimit filter*

*Description*:
When using [RateLimit](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rate_limit_filter) envoy filter:
```
type.googleapis.com/envoy.extensions.filters.http.ratelimit.v3.RateLimit
```
we might want to set which http status code should be returned in case of communication failures / timeout etc.

Currently, there is no option to specify what should be returned, and by default, `500` response code is returned when there is an error in ratelimit-service or it cannot be reached. 

The response code returned by the Envoy should be configurable, there is such option in `ExtAuthz` filter and it is called `status_on_error`. We should add something similar to `RateLimit` filter.

*Relevant Links*:
[ExtAuthz](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto) - check `status_on_error`

**Comment by ggreenway** [comment]
cc @esmet @mattklein123

**Comment by StarryVae** [comment]
👀 if needed, i will try to pick it up, thanks.

**Comment by AleksanderBrzozowski** [comment]
@ggreenway @esmet @mattklein123 Could you take a look at this issue? :)

**Comment by ggreenway** [comment]
Seems reasonable, PRs welcome

**Comment by StarryVae** [comment]
this issue has not been resolved yet, can we reopen it? and i will submit a PR, thanks. cc @ggreenway @mattklein123

**Comment by StarryVae** [comment]
done, this issue can be closed.

**Comment by ggreenway** [comment]
@StarryVae In the future, in your PR if you add `Fixes #27423` github will auto-close the related issue when the PR is merged

**Comment by StarryVae** [comment]
> @StarryVae In the future, in your PR if you add `Fixes #27423` github will auto-close the related issue when the PR is merged

ok, thanks!
